base/io.jl

```diff
@@ -682,7 +682,7 @@ function read!(s::IO, a::Array{UInt8})
     return a
 end
 
-function read!(s::IO, a::Array{T}) where T
+function read!(s::IO, a::Union{Array{T}, SubArray{T,N,<:Array{T}} where N}) where T
```

Comment:
This should address the first half of Yichao's comment:

```suggestion
function read!(s::IO, a::Union{Array{T}, FastContiguousSubArray{T,N,<:Array{T}} where N}) where T
```

Reply:
Cool, thank you very much, I don't think I would have found this type very quickly!